flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSourceSinkFactory.java

```diff
@@ -0,0 +1,283 @@
+		.key("write.flush.interval")
+		.longType()
+		.defaultValue(0L)
+		.withDescription("-- optional, flush interval mills, over this time, asynchronous threads will flush data. The " +
+			"default value is 0, which means no asynchronous flush thread will be scheduled.");
+	private static final ConfigOption<Integer> WRITE_MAX_RETRIES = ConfigOptions
+		.key("write.max-retries")
+		.intType()
+		.defaultValue(3)
+		.withDescription("-- optional, max retry times if writing records to database failed.");
+
+	@Override
+	public DynamicTableSink createDynamicTableSink(Context context) {
+		final FactoryUtil.TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
+		helper.validate();
+		final JdbcOptions jdbcOptions = getJdbcOptions(helper.getOptions());
+		final DataType rowDataType = context.getCatalogTable().getSchema().toPhysicalRowDataType();
+		final TableSchema formatSchema = TableSchemaUtils.getPhysicalSchema(context.getCatalogTable().getSchema());
+		final DataType[] fieldDataTypes = formatSchema.getFieldDataTypes();
+
+		return new JdbcDynamicTableSink(
+			jdbcOptions,
+			getJdbcExecutionOptions(helper.getOptions()),
+			getJdbcDmlOptions(jdbcOptions, context.getCatalogTable().getSchema()),
+			rowDataType,
+			fieldDataTypes);
+	}
+
+	@Override
+	public DynamicTableSource createDynamicTableSource(Context context) {
+		final FactoryUtil.TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
+		helper.validate();
+		TableSchema formatSchema = TableSchemaUtils.getPhysicalSchema(context.getCatalogTable().getSchema());
+
+		int[] selectFields = new int[formatSchema.getFieldNames().length];
+		for (int i = 0; i < selectFields.length; i++) {
+			selectFields[i] = i;
+		}
+		return new JdbcDynamicTableSource(
+			getJdbcOptions(helper.getOptions()),
+			getJdbcReadOptions(helper.getOptions()),
+			getJdbcLookupOptions(helper.getOptions()),
+			formatSchema,
+			selectFields);
+	}
+
+	private JdbcOptions getJdbcOptions(ReadableConfig readableConfig) {
+		final String url = readableConfig.get(URL);
+		final JdbcOptions.Builder builder = JdbcOptions.builder()
+			.setDBUrl(url)
+			.setTableName(readableConfig.get(TABLE))
+			.setDialect(JdbcDialects.get(url).get());
+
+		readableConfig.getOptional(DRIVER).ifPresent(builder::setDriverName);
+		readableConfig.getOptional(USERNAME).ifPresent(builder::setUsername);
+		readableConfig.getOptional(PASSWORD).ifPresent(builder::setPassword);
+		return builder.build();
+	}
+
+	private JdbcReadOptions getJdbcReadOptions(ReadableConfig readableConfig) {
+		final Optional<String> partitionColumnName = readableConfig.getOptional(READ_PARTITION_COLUMN);
+		final Optional<Long> partitionLower = readableConfig.getOptional(READ_PARTITION_LOWER_BOUND);
+		final Optional<Long> partitionUpper = readableConfig.getOptional(READ_PARTITION_UPPER_BOUND);
+		final Optional<Integer> numPartitions = readableConfig.getOptional(READ_PARTITION_NUM);
```

Comment:
You can put these 3 in the `partitionColumnName.isPresent()` branch block, and use `config.get(..)` to avoid the warning.